PBTA Histologies: Independent sample base (2 of N)

kgaonkar6 opened the pull request:
<!--Hi there, thanks for your contribution! Please take a moment to fill out this template to facilitate the review of your pull request.-->

### Purpose/implementation Section

#### What scientific question is your analysis addressing?
Rna summary modules needed for subtyping need to be updated because of the 8 new stranded files in v18. In this PR  I'm updating independent-samples module to read from pbta-histologies-base.tsv which will be used to generate RNA summary files and run molecular subtyping modules.

| Module |  Reason |Brief Description | output
|--------|----------|--------|------
| [`independent-samples`](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/independent-samples) | adding 8 samples #749 and used in subtyping  | Generates independent specimen lists for WGS/WXS samples updated in  #795 and #797 and [comment](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/750#issuecomment-691526896)| `results/independent-specimens.wgs.primary.tsv` (included in data download) <br> `results/independent-specimens.wgs.primary-plus.tsv` (included in data download) <br> `results/independent-specimens.wgswxs.primary.tsv` (included in data download) <br> `results/independent-specimens.wgswxs.primary-plus.tsv` (included in data download) <br> `results/independent-specimens.rnaseq.primary-plus-stranded.tsv`(included in data download) <br> `results/independent-specimens.rnaseq.primary-plus-polya.tsv` (included in data download)
|


#### What GitHub issue does your pull request address?
#861 


### Directions for reviewers. Tell potential reviewers what kind of feedback you are soliciting.

#### Which areas should receive a particularly close look?
I've added a condition to run independent-samples modules with pbta-histologies-base.tsv in run-independent-samples.sh
Most of the different samples selected in the output files is from random selection since I didn't update the actual code generating the list.
Other modules that need to be rerun for subtyping are in #861 



But some changes occurred because of different tumor_descriptors in v18 
PT_ID | BS_ID | Change
--- | --- | ---
PT_1VQWQ0TC |	BS_QXZTRMWM  | was Initial now Recurrence

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
Yes


### Results

#### What types of results are included (e.g., table, figure)?
tables


#### What is your summary of the results?
8 new samples added to stranded RNA independent sample list and updated tumor_descriptor used for WGS/WXS independent sample list


#### Reproducibility Checklist

<!-- Check all those that apply or remove this section if it is not applicable.-->

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

<!-- Please review https://github.com/AlexsLemonade/OpenPBTA-analysis#documenting-your-analysis -->

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [ ] The analytical code is documented and contains comments.

<!-- IF YOUR PULL REQUEST IS A DATA RELEASE, PLEASE REMOVE THE [HTML COMMENT TAG](https://html.com/tags/comment-tag/) FROM THE SECTION BELOW AND COMPLETE THE CHECKLIST-->

<!-- 
### Data Release Checklist
- [ ] Is the table in doc/data-file-descriptions.md up to date?
- [ ] Is doc/data-format.md up to date?
- [ ] Is doc/release-notes.md up to date?
- [ ] Is download-data.sh up to date?
- [ ] Was download-data.sh tested and did it complete without error?
-->

jharenza reviewed (commented):
@kgaonkar6 It looks like the files from `collapse-rnaseq` were also included in this PR - were you staggering these PRs or should these not be here?

jharenza reviewed (approved):
Works as expected. Using the v18 data download, I ran with 
```
RUN_FOR_SUBTYPING=${BASE_SUBTYPING:-1}
```

and got the same file dims as you did, but did not compare directly because of the random selection.

Just check out my comment about about whether you should remove the files from the previous PR submitted.

kgaonkar6 commented:
These are staggered, I used instructions in https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/CONTRIBUTING.md#creating-stacked-pull-requests let me know it I missed something. 

hmm CI seems to error out because it doesn't have a file named pbta-histologies-base.tsv should I add that to the CI subset files?

jharenza commented:
> These are staggered, I used instructions in https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/CONTRIBUTING.md#creating-stacked-pull-requests let me know it I missed something.
> 
> hmm CI seems to error out because it doesn't have a file named pbta-histologies-base.tsv should I add that to the CI subset files?

Ok great. No, I don't think there is a need for that right now because these PRs will not be merged until #857 goes in, which depends on CI file updates by @jaclyn-taroni and/or someone at CCDL.

jharenza reviewed (approved):
This looks good @kgaonkar6 - will you also update the READMEs for all of the modules to which you are adding the subtyping base run to note the option?

Thanks!

kgaonkar6 commented:
Hi @jharenza  just updated the README

jharenza reviewed (approved):
thanks for the updates! looks good.

cansavvy commented on analyses/independent-samples/run-independent-samples.sh:
```diff
@@ -22,3 +28,14 @@ Rscript 02-generate-independent-rnaseq.R \
 
+else 
+Rscript 01-generate-independent-specimens.R \
+  -f ../../data/pbta-histologies-base.tsv \
```
It looks like the only thing that is different is whether `../../data/pbta-histologies-base.tsv`or `../../data/pbta-histologies.tsv` is used, so a shorter way to do this is to use a smaller if/then specify a `HISTOLOGIES_FILE` variable and supply that to the `-f` and `--histology_file` options. Then you can get rid of this larger if/then and not have to repeat these Rscript calls.

kgaonkar6 replied:
ah that's a great idea, I'll update that, thanks!

cansavvy commented on analyses/independent-samples/README.md:
````diff
@@ -25,11 +25,18 @@ independent-specimens.rnaseq.primary-plus-stranded.tsv
 
+use BASE_SUBTYPING=1 to run this module using the pbta-histologies-base.tsv from data folder while running molecular-subtyping modules for release.
+```sh
+BASE_SUBTYPING=1 ../analyses/independent-samples/run-independent-samples.sh 
````
It's unclear to me which histologies file should be used for the circle CI, but right now it doesn't have its own variable supplied and seems to be failing. 

You can follow the OpenPBTA instructions here https://github.com/AlexsLemonade/OpenPBTA-analysis#passing-variables-only-in-ci to add an explicit call to the circle CI file so its clear which file should be being used.

jharenza replied:
Hi @cansavvy! Thanks for your comments. This sequence of PRs will require #849 to be merged and #871 to be generated, as they are utilizing the `pbta-histologies-base.tsv` file. I think the idea is for these to be reviewed, as multiple of the files generated from these modules will be included in the release, then for us to merge #849, and have all of these run in CI for confirmation they pass.

kgaonkar6 replied:
Also I wanted to confirm that in the v18 release data folder we will have eventually have from #870  pbta-histologies.tsv so it should be able to run through .circleci/config.yml for checks as is, right? Or do I add the following to .circleci/config.yml?
```
BASE_SUBTYPING=1 ../analyses/independent-samples/run-independent-samples.sh 
```

cansavvy replied:
Yes, but according to the instructions, you will want to start the variable with `OPENPBTA_` like the examples show: https://github.com/AlexsLemonade/OpenPBTA-analysis#passing-variables-only-in-ci 
```
Any environment variables prefixed with OPENPBTA_ are passed to the specified shell script. Environment variables without this prefix are not passed.
```

kgaonkar6 replied:
ah got it, I didn't realize we will be running the BASE_SUBTYPING=1 runs in CI. But since we will I'm going to update the name for the variable as you've suggested here and re-run

kgaonkar6 replied:
Updated to OPENPBTA_BASE_HISTOLOGY variable

cansavvy reviewed (commented):
I think this mostly looks good, I just have two comments that should probably be addressed (and might help resolve that circle CI error).

kgaonkar6 commented:
thanks for review @cansavvy @jharenza , latest commit was from a re-run with updated base histology(minor change to Normal sample rows) file which doesn't change any results files.

kgaonkar6 commented:
updated the run script with a simple if else for HISTOLOGY_FILE assignment instead of the previous longer code update as suggested by @cansavvy . I also updated reading `results/independent-specimens.wgswxs.primary-plus.tsv` instead of the data folder since we want to read the latest file generated for WGSWXS specimens from results folder.

jaclyn-taroni commented:
It looks to me like all of @cansavvy's comments have been address since she last reviewed. I will merge if/when CI passes.
